geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -268,6 +268,161 @@ impl IntersectionMatrix {
             && self.0[CoordPos::OnBoundary][CoordPos::Outside] == Dimensions::Empty
     }
 
+    /// Tests whether this matrix matches at least one of `[T*F**F***]`, `[*TF**F***]`, `[**FT*F***]`, or `[**F*TF***]`
+    ///
+    /// - `a` is covered by `b` (extends `is_within`): geometry `a` lies in `b`. OR
+    /// - At least **one** point of `a` lies in `b`, and **no** point of `a` lies in the **exterior** of `b` OR
+    /// - **Every** point of `a` is a point of (the **interior** or **boundary** of) `b`
+    ///
+    /// returns `true` if the first geometry is covered by the second.
+    ///
+    /// ```
+    /// use geo_types::{Polygon, polygon};
+    /// use geo::relate::Relate;
+    ///
+    /// let poly1 = polygon![
+    ///     (x: 125., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 160., y: 160.),
+    ///     (x: 125., y: 179.),
+    /// ];
+    /// let poly2 = polygon![
+    ///     (x: 124., y: 182.),
+    ///     (x: 106., y: 146.),
+    ///     (x: 162., y: 159.),
+    ///     (x: 124., y: 182.),
+    /// ];
+    ///
+    /// let intersection = poly1.relate(&poly2);
+    /// assert_eq!(intersection.is_coveredby(), true);
+    /// ```
+    #[allow(clippy::nonminimal_bool)]
```

Comment:
Clippy has suggested simplified boolean logic but I want to maintain what's here as it's easy to check against the documented relations.